src/utils/useSessionState.ts

```diff
@@ -0,0 +1,105 @@
+import { Dispatch, SetStateAction, useCallback, useEffect, useMemo, useRef, useState } from "react";
```

Comment:
We should try to write automated tests were appropriate. I think a util function like this is pretty straightforward to test.

Reply:
Done